### PR TITLE
fix(pie-link): DSW-1185 address design reviews

### PR DESCRIPTION
**Pull request by raoufswe**
## Describe your changes (can list changeset entries if preferable)
[Fixed] - When the `isStandalone` prop is `false`, underline can only be default and not reversed.

[Fixed] - When the `isStandalone` prop is `false`, icon's cannot be shown.

[Fixed] - When the link is small, the icon doesn't seem to be vertically centered with the text.

[Fixed] - The spacing between the text and icon should be fixed / hugging the length of the text.

[Fixed] - focus styles for the link component

## Author Checklist (complete before requesting a review)
- [ ] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests
- [ ] If it is a `PIE Docs` change, I have reviewed the Docs site preview
- [ ] If it is a component change, I have reviewed the Storybook preview
- [ ] If there are visual test updates, I have reviewed them properly before approving

### Reviewer 1
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### Reviewer 2
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### CODE REVIEW

File: .changeset/clean-deers-arrive.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+---
+"@justeattakeaway/pie-link": minor
+"pie-storybook": minor
+---
+
+[Fixed] - When the `isStandalone` prop is `false`, underline can only be default and not reversed.
+
+[Fixed] - When the `isStandalone` prop is `false`, icon's cannot be shown.
+
+[Fixed] - When the link is small, he icon doesn't seem to be vertically centered with the text.
```
Comment (ashleynolan) [inline .changeset/clean-deers-arrive.md:10]:
```suggestion
[Fixed] - When the link is small, the icon doesn't seem to be vertically centered with the text.
```

File: apps/pie-storybook/stories/pie-link.stories.ts [hunk 2]
```diff
@@ -55,15 +55,15 @@ const linkStoryMeta: LinkStoryMeta = {
             },
         },
         underline: {
-            description: 'Set the underline behavior of the link.',
+            description: 'Set the underline behavior of the link.<br /><br /> the `reverse` type can only be used if `isStandalone` is `true`',
```
Comment (ashleynolan) [inline apps/pie-storybook/stories/pie-link.stories.ts:58]:
```suggestion
            description: 'Set the underline behavior of the link.<br /><br />The `reverse` type can only be used if `isStandalone` is `true`',
```

File: apps/pie-storybook/stories/pie-link.stories.ts [hunk 3]
```diff
@@ -55,15 +55,15 @@ const linkStoryMeta: LinkStoryMeta = {
...
             control: 'select',
             options: underlineTypes,
             defaultValue: {
                 summary: 'default',
             },
         },
         iconPlacement: {
-            description: 'Show a leading/trailing icon.<br /><br />To use this with pie-link, you can pass an icon into the `icon` slot.',
+            description: 'Show a leading or trailing icon.<br /><br />To use this with pie-link, you can pass an icon into the `icon` slot.<br /><br /> Can only be used if `isStandalone` is `true`',
```
Comment (ashleynolan) [inline apps/pie-storybook/stories/pie-link.stories.ts:66]:
```suggestion
            description: 'Show a leading or trailing icon.<br /><br />To use this with pie-link, you can pass an icon into the `icon` slot.<br /><br />Can only be used if `isStandalone` is `true`',
```

File: packages/components/pie-link/src/defs.ts [hunk 4]
```diff
@@ -27,7 +27,8 @@ export interface LinkProps {
      */
     size: typeof sizes[number];
     /**
-     * What underline behavior the link should have such as default or reversed
+     * What underline behavior the link should have such as default or reversed.
```
Comment (ashleynolan) [inline packages/components/pie-link/src/defs.ts:30]:
```suggestion
     * Defines what underline behavior the link should have, such as default or reversed.
```

File: packages/components/pie-link/src/index.ts [hunk 5]
```diff
@@ -73,16 +73,16 @@ export class PieLink extends LitElement implements LinkProps {
 
     /**
      * Renders the link content.
-     *
+     * Icons should render only if the link is a block element
```
Comment (ashleynolan) [inline packages/components/pie-link/src/index.ts:76]:
```suggestion
     * Icons should only render if the link is a block element
```

File: packages/components/pie-link/src/link.scss [hunk 6]
```diff
@@ -65,11 +70,9 @@
 
     &[underline='default'] {
         /* Same as default styles */
-
-        @include link-interactive-states('default');
     }
 
-    &[underline='reversed'] {
+    &[underline='reversed'][isStandalone] {
```
Comment (ashleynolan) [inline packages/components/pie-link/src/link.scss:75]: In the changelog it says "When the `isStandalone` prop is `false`, underline can only be default and not reversed."

But this looks like this applies the styles when underline is `reversed` and `isStandalone` is true?

Comment (raoufswe) [reply packages/components/pie-link/src/link.scss:75]: aha sorry bad wording, I will update the changelog. 
but basically, the reverse type must be used only if the `isStandalone` is true. (only if the link is a block element as reversed + inline will be confusing for users).

File: apps/pie-storybook/stories/pie-link.stories.ts [hunk 7]
```diff
@@ -55,15 +55,15 @@ const linkStoryMeta: LinkStoryMeta = {
             },
         },
         underline: {
-            description: 'Set the underline behavior of the link.',
+            description: 'Set the underline behavior of the link.<br /><br /> The `reverse` type can only be used if `isStandalone` is `true`',
```
Comment (webbertakken) [inline apps/pie-storybook/stories/pie-link.stories.ts:58]:
```suggestion
            description: 'Set the underline behavior of the link.<br /><br />The `reverse` type can only be used if `isStandalone` is `true`',
```

File: apps/pie-storybook/stories/pie-link.stories.ts [hunk 8]
```diff
@@ -55,15 +55,15 @@ const linkStoryMeta: LinkStoryMeta = {
...
             control: 'select',
             options: underlineTypes,
             defaultValue: {
                 summary: 'default',
             },
         },
         iconPlacement: {
-            description: 'Show a leading/trailing icon.<br /><br />To use this with pie-link, you can pass an icon into the `icon` slot.',
+            description: 'Show a leading or trailing icon.<br /><br />To use this with pie-link, you can pass an icon into the `icon` slot.<br /><br />Can only be used if `isStandalone` is `true`',
```
Comment (webbertakken) [inline apps/pie-storybook/stories/pie-link.stories.ts:66]:
```suggestion
            description: 'Show a leading or trailing icon.<br /><br />To use this with pie-link, you can pass an icon into the `icon` slot.<br /><br />Can only be used if `isStandalone` is set to `true`',
```

File: packages/components/pie-link/README.md [hunk 9]
```diff
@@ -62,15 +62,15 @@ import { PieLink } from '@justeattakeaway/pie-link/dist/react';
 | tag           | `String`  | `a`         | The rendered HTML element of the link, one of `tags` – `a`, `button`                        |
 | variant       | `String`  | `default`   | Variant of the link, one of `variants` – `default`, `high-visibility`, `inverse`         |
 | size          | `String`  | `medium`    | Size of the link, one of `sizes` – `medium`, `small`                                          |
-| underline          | `String`  | `default`    | The underline behavior of the link, one of `underlineTypes` – `default`, `reversed`                                          |
+| underline          | `String`  | `default`    | The underline behavior of the link, one of `underlineTypes` – `default`, `reversed`. The `reverse` type can only be used if `isStandalone` is `true`                                          |
```
Comment (webbertakken) [inline packages/components/pie-link/README.md:65]:
```suggestion
| underline          | `String`  | `default`    | The underline behavior of the link, one of `underlineTypes` – `default`, `reversed`. The `reverse` type can only be used if `isStandalone` is set to `true`                                          |
```

File: packages/components/pie-link/README.md [hunk 10]
```diff
@@ -62,15 +62,15 @@ import { PieLink } from '@justeattakeaway/pie-link/dist/react';
...
 | target        | `String`  | `undefined` | Native html `target` attribute                                                                     |
 | type          | `String`  | `submit`    | Native html `type` attribute if the tag is set to `button`                                       |
 | isBold        | `Boolean` | `false`     | If `true`, sets the link text bold                                                                 |
 | isStandalone  | `Boolean` | `false`     | If `true`, sets the link as a block element                                                        |
 | hasVisited    | `Boolean` | `false`     | If `true`, the link will apply the styles for the visited state                                    |
-| iconPlacement | `String`  | `leading`   | Icon placements of the icon slot, if provided, one of `iconPlacements` - `leading`, `trailing` |
+| iconPlacement | `String`  | `leading`   | Icon placements of the icon slot, if provided, one of `iconPlacements` - `leading`, `trailing`. Can only be used if `isStandalone` is `true` |
```
Comment (webbertakken) [inline packages/components/pie-link/README.md:73]:
```suggestion
| iconPlacement | `String`  | `leading`   | Icon placements of the icon slot, if provided, one of `iconPlacements` - `leading`, `trailing`. Can only be used if `isStandalone` is set to `true` |
```

File: packages/components/pie-link/src/index.ts [hunk 11]
```diff
@@ -73,16 +73,16 @@ export class PieLink extends LitElement implements LinkProps {
 
     /**
      * Renders the link content.
-     *
+     * Icons should only render if the link is a block element
```
Comment (webbertakken) [inline packages/components/pie-link/src/index.ts:76]: I'd suggest to leave away words that are already implied by the context like "should" and "link" and leave away words that are technical by choosing "display" over "render". This makes it easier to understand for new developers.

```suggestion
     * Icons are only shown in block elements
```

**Review by webbertakken — APPROVED** [review]
Nice!

**Review by dandel10n — APPROVED** [review]

**Review by webbertakken — APPROVED** [review]

**Review by dandel10n — APPROVED** [review]

**Pull request merged**